README.md

```diff
@@ -2,45 +2,153 @@
-- Retrieving [artifacts](https://github.com/open-component-model/ocm-spec/blob/main/doc/glossary.md#artifact) from OCM repositories
-- Making individual resources from the OCM component available within the cluster
-- Performing localization and configuration
+Apply the `Resource`:
 
-Snapshots are used to pass blobs between controllers and snapshots are stored in the internal registry managed by the `ocm-controller`.
+```bash
+kubectl apply -f resource.yaml
+```
 
-![ocm controller architecture](./docs/diagrams/ocm-controllers-reconcile-arch.png)
+Create a Flux `Kustomization` to apply the `Resource` and save it to a file named `resource_kustomization.yaml`:
+
+```yaml
+# resource_kustomization.yaml
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
+kind: Kustomization
+metadata:
+  name: podinfo
+  namespace: ocm-system
+spec:
+  interval: 10m0s
+  path: ./
+  prune: true
+  targetNamespace: default
+  sourceRef:
+    kind: OCIRepository
+    name: podinfo
+    namespace: ocm-system
+```
 
-## Testing
+Apply the `Kustomization`:
 
-In order to test, run the manager locally with a kind cluster present on the system.
+```
+kubectl apply -f resource_kustomization.yaml
+```
 
-- apply all CRDs from the crd/base folder
-- run the controller:
+View the deployment spinning up:
 
-```console
-make
-./bin/manager -zap-log-level 4
-```
+`kubectl get po -n default`
+
+### What just happened?
+
+We used `ComponentVersion` to retrieve the `phoban.io/podinfo` component from an remote OCM repository (`ghcr.io/phoban01`). We then fetched a resource from this component using the `Resource` CRD.
+
+The `ocm-controller` fetched this resource and created a `Snapshot` containing the contents of the OCM resource. This `Snapshot` is a Flux compatible OCI image which is stored in a registry managed by the `ocm-controller`.
 
-- apply all the sample objects
+Because the `Snapshot` is Flux compatible we asked the `ocm-controller` to create a corresponding Flux source for the `Snapshot`. This means that we can then use a Flux `Kustomization` resource to apply the `Snapshot` to the cluster.
 
-To trigger the `reconcile` loop ( because manually updating the status doesn't trigger it ) run something like this:
+The following diagram illustrates the flow:
 
-```console
-kubectl annotate --overwrite ... "trigger/reconcile=$(date -u+%)"
+```mermaid
+  graph TD;
+      ComponentVersion --> Resource --> Snapshot;
+      OCIRepository --> Snapshot;
+      subgraph flux;
+          OCIRepository --> Kustomization --> Deployment;
+    end
 ```
 
+## Elements
+
+### ComponentVersion
+
+Retrieves a `ComponentVersion` from an OCM repository. Handles authentication with the repository and optionally verifies the component using provided signatures.
+
+### Resource
+
+Makes a resource available within the cluster as a snapshot.
+
+### Localization
+<!-- TODO: add a dedicated doc on Localization -->
+Localizes a resource using the specified configuration resource.
```

Comment:
Just add a tiny example of an image exchange to hammer the point home what Localization means.

Reply:
I'm going to add a dedicated doc on both Localization and Configuration, there's a bunch of explaining about both what the mean and how to do them. Won't add it now however because we'll likely change the ConfigData name soon.